bignum/src/commonMain/kotlin/com/ionspin/kotlin/bignum/integer/BigInteger.kt

```diff
@@ -47,6 +47,12 @@ class BigInteger internal constructor(wordArray: WordArray, val sign: Sign) : Bi
     constructor(short: Short) : this(arithmetic.fromShort(short), determinSignFromNumber(short))
     constructor(byte: Byte) : this(arithmetic.fromByte(byte), determinSignFromNumber(byte))
 
+    init {
```

Comment:
I agree that BigIntegers with zero magnitude should have Sign.ZERO, but adding this breaks a lot of things, just from one test run there were 12 failing tests.

 One way to fix this is to check wherever the code calls the constructor with sign parameter and check if the magnitude is zero and then correct the sign. 

Another way which should be easier, would be to change the constructor to something like:
`internal constructor(wordArray: WordArray, requestedSign: Sign)`
and then check the magnitude before setting the correct sign, i.e.:
```
    internal val sign: Sign = if (isResultZero(magnitude)) {
        Sign.ZERO
    } else {
        requestedSign
    }
```

Reply:
That is an interesting idea, but still allows initialising BigInteger with Sign.ZERO and a non-zero magnitude
so i kept the assertion for that case. All tests have passed.

Reply:
So to sum-up:
BigInteger(<zero magnitude>, <non-zero sign>) -> sign will be fixed to zero
BigInteger(<non-zero magnitude>, Sign.ZERO) -> assertion throws an error

Reply:
Good point!